src/Issues.ts

```diff
@@ -21,24 +32,107 @@ export class Issues{
+        try {
+            let meta = await this._getMetadata();
+            let apiUrl = api + '/repos/' + meta.user + '/' + meta.repo + '/labels';
+            let res = await axios.post(apiUrl, {
+                "name": label
+            }, {
+                    auth: {
+                        username: meta.user,
+                        password: config.getToken()
+                    }
+                });
+            return res;
+        } catch (e) {
+            console.error(e);
+            return false;
+        }
+    }
+
+    public async getIssues() {
+        try {
             let meta = await this._getMetadata();
             let apiUrl = api + '/repos/' + meta.user + '/' + meta.repo + '/issues';
             let res = await axios.get(apiUrl);
             return res.data;
-        }catch(e){
+        } catch (e) {
             console.error(e);
-            return false ;
+            return false;
         }
     }
 
-    public async addLabel(label:string,number:number){
-        try{
+    public async addLabel(label: string, number: number) {
+        try {
+            if (config.getToken().length < 1) {
+                window.showInformationMessage('please add a token or set the password for adding labels');
+                return false;
+            }
+
             let meta = await this._getMetadata();
             let apiUrl = api + '/repos/' + meta.user + '/' + meta.repo + '/issues/' + number + '/labels';
-            let res = await axios.post(apiUrl,label);
-            console.log(res);
+            let checkResult = await this._checkLabel(label);
+
+            if (checkResult == -1) {
+                let res = await this._createLabel(label);
+                if (res.status != 201) {
+                    return false;
+                }
+            }
+
+            let res = await axios.post(apiUrl, [label], {
+                auth: {
+                    username: meta.user,
+                    password: config.getToken()
+                }
+            });
+
+            if (res.status == 200) {
+                return true;
+            } else {
+                return false;
+            }
+        } catch (e) {
+            console.error(e);
+            return false;
+        }
+    }
+
+    public async removeLabel(label: string, number: number) {
+        try{
+            if (config.getToken().length < 1) {
+                window.showInformationMessage('please add a token or set the password for adding labels');
+                return false;
+            }
+            let meta = await this._getMetadata();
+            let apiUrl = api + '/repos/' + meta.user + '/' + meta.repo + '/issues/' + number + '/labels/' + label;
+            let res = await axios.delete(apiUrl, {
+                auth: {
+                    username: meta.user,
+                    password: config.getToken()
+                }
+            });
+            if(res.status == 200){
+                return true;
```

Comment:
this can be rewritten as 

`return res.status === 200` without using if statements right?